### PR TITLE
THRIFT-5129: Fix swift TSocketTransport on Mac

### PR DESCRIPTION
**Pull request by taiyangc**
Fix TSocketTransport of Swift client not working for mac.

Not sure why it was `in_addr_t(0)` in the first place. I tested locally on my mac and the client is working again. Let me know what other changes/clarifications are needed.

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Comment by taiyangc** [comment]
@Jens-G I think Thrift has been falling behind with major language support for the past few years. I understand that certain languages have breaking changes that are hard to follow up, but as a basic rule, the simplest protocols should not break (which is not the case in my current PR).

I hope to get some attention (not just from you personally), but as a whole, how Thrift can evolve forward and combat other standards/frameworks such as grpc/protobuf. It's not like I prefer switching to them, but if basic language support isn't up to par, there is no reason for developers to stay with Thrift. I hope I expressed my concerns concise and to point, but if I am wrong, I am very open to suggestions and criticism.

**Comment by janosvitok** [comment]
This fixes https://issues.apache.org/jira/browse/THRIFT-5129. @taiyangc, you may want to edit PR description to mention this issue (and also rename it to THRIFT-5129: ...)

Please check my comment in the issue (regarding UInt16(truncatingIfNeeded: port).bigEndian). It was needed for SocketServer for iOS app, but I didn't check why the client compiles without it...

**Comment by janosvitok** [comment]
Also, regarding the state of swift - you may want to check issues files with [component Swift - Library and Swift - Compiler.](https://issues.apache.org/jira/browse/THRIFT-5195?jql=project%20%3D%20THRIFT%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20AND%20component%20in%20(%22Swift%20-%20Compiler%22%2C%20%22Swift%20-%20Library%22)) and also PR's here. There are a few fixes not merged yet.

There was a recent [call for maintainers](http://mail-archives.apache.org/mod_mbox/thrift-dev/202004.mbox/%3C8968F4A70F5F4983AB1072F381C06ADA%40HAGGIS%3E), that you may want to consider.

**Comment by taiyangc** [comment]
@janosvitok Thanks, updated.
